python/src/aiconfig/editor/server/server.py

```diff
@@ -161,25 +170,260 @@ def create() -> FlaskResponse:
             return HttpResponseWithAIConfig(message=f"Failed to create AIConfig: {e}", code=400, aiconfig=None).to_flask_format()
 
 
+@app.route("/api/test_streaming", methods=["POST"])
+def test_streaming():
+    EXCLUDE_OPTIONS = {
+        "prompt_index": True,
+        "file_path": True,
+        "callback_manager": True,
+    }
+    state = get_server_state(app)
+    aiconfig = state.aiconfig
+    request_json = request.get_json()
+    num_stream_steps: int = request_json.get("num_stream_steps", 10)
+    prompt_name: str = request_json.get("prompt_name", "get_activities")
+    params = request_json.get("params", {})
+    stream = request_json.get("stream", True)
+    print(f"{num_stream_steps=}")
+    print(f"{type(num_stream_steps)=}")
+
+    # Define stream callback and queue object for streaming results
+    output_text_queue = AsyncQueueIterator()
+    def update_output_queue(data, _accumulated_data, _index) -> None:
+        should_end_stream = data == STOP_STREAMING_SIGNAL
+        output_text_queue.put(data, should_end_stream)
+    inference_options = InferenceOptions(
+        stream=stream,
+        stream_callback=update_output_queue,
+    )
+
+    def generate(num_stream_steps: int):
+        aiconfig_json: str | None = None
+        prompt: Prompt = state.aiconfig.get_prompt(prompt_name)
+        def run_async_config_in_thread():
+            asyncio.run(aiconfig.run(
+                    prompt_name=prompt_name,
+                    params=params,
+                    run_with_dependencies=False,
+                    options=inference_options,
+                )
+            )
+            output_text_queue.put(STOP_STREAMING_SIGNAL)
+        t = threading.Thread(target=run_async_config_in_thread)
+        t.start()
+
+
+        # Need to wait until streamer has at least 1 stream
+        # def wait_for_stream():
+        #     while output_text_queue.empty():
+        #         time.sleep(0.1)
```

Comment:
This is what I was having trouble with. Gradio is slow enough that by the time we end up at this if-loop check the aiconfig has already started running and we have output in the queue

Reply:
NM, this could jsut be perhaps because Queue object is blocking while asyncio.Queue is non-blocking. I realized since I have this within `generator` function which is not async I can just use original queue object